playbooks/ansible-test-units-base/post.yaml

```diff
@@ -1,12 +1,53 @@
 ---
 - hosts: controller
+  vars:
+    ansible_test_location: "{{ ansible_user_dir }}/{{ zuul.projects[ansible_collections_repo].src_dir }}"
+    ansible_coverage_report: "ansible-test coverage html"
+    ansible_test_venv_path: "~/venv"
+    ansible_test_collection_dir: "~/.ansible/collections/ansible_collections"
   tasks:
+    - name: Get collection namespace
+      args:
+        chdir: "{{ ansible_test_location }}"
+        executable: /bin/bash
+      shell: "cat galaxy.yml | {{ ansible_test_venv_path }}/bin/yq -y .namespace | tail -n +1 | head -1"
+      register: _collection_namespace
+
+    - name: Get collection name
+      args:
+        chdir: "{{ ansible_test_location }}"
+        executable: /bin/bash
+      shell: "cat galaxy.yml | {{ ansible_test_venv_path }}/bin/yq -y .name | tail -n +1 | head -1"
+      register: _collection_name
+
+    - name: Setup location of project for tests
+      set_fact:
+        test_location: "{{ ansible_test_collection_dir }}/{{ _collection_namespace.stdout }}/{{ _collection_name.stdout }}"
+
+    - name: Coverage report for unit tests
+      args:
+        chdir: "{{ test_location }}"
+        executable: /bin/bash
+      environment: "{{ ansible_test_environment | default({}) }}"
+      shell: "source {{ ansible_test_venv_path }}/bin/activate; {{ ansible_coverage_report }}"
+
+    - name: Ensure controller directory exists
+      file:
+        path: "{{ ansible_user_dir }}/zuul-output/logs/controller"
+        state: directory
+
+    - name: Create a symbolic link to the coverage directory
```

Comment:
k, we don't actually need to do a symlink here, I was suggesting we keep that in run.yaml, for now let just do

```
cp -a {{ test_location }}/tests/output/reports/coverage {{ ansible_user_dir }}/zuul-output/logs/controller/
```